interface/forms/soap/C_FormSOAP.class.php

```diff
@@ -1,40 +1,44 @@
 <?php
 
-/**
+/*
  * soap form
- *
  * @package   OpenEMR
  * @link      http://www.open-emr.org
  * @author    Brady Miller <brady.g.miller@gmail.com>
+ * @author    Sherwin Gaddis <sherwingaddis@gmail.com>
  * @copyright Copyright (c) 2019 Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (c) 2022 Sherwin Gaddis <sherwingaddis@gmail.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
 
 require_once($GLOBALS['fileroot'] . "/library/forms.inc.php");
 require_once("FormSOAP.class.php");
+require_once("Bootstrap.php");
 
 use OpenEMR\Common\Csrf\CsrfUtils;
 
 class C_FormSOAP extends Controller
 {
-    var $template_dir;
+    private \Twig\Environment $template;
 
-    function __construct($template_mod = "general")
+    public function __construct()
     {
-        parent::__construct();
-        $this->template_mod = $template_mod;
-        $this->template_dir = dirname(__FILE__) . "/templates/";
-        $this->assign("FORM_ACTION", $GLOBALS['web_root']);
-        $this->assign("DONT_SAVE_LINK", $GLOBALS['form_exit_url']);
-        $this->assign("STYLE", $GLOBALS['style']);
-        $this->assign("CSRF_TOKEN_FORM", CsrfUtils::collectCsrfToken());
+        $twig = new Bootstrap();
+        $this->template = $twig->twigEnv();
     }
 
     function default_action()
     {
-        $form = new FormSOAP();
-        $this->assign("data", $form);
-        return $this->fetch($this->template_dir . $this->template_mod . "_new.html");
+         $form = new FormSOAP();
+        return $this->template->render(
```

Comment:
hopefully can do it all with following one liner (or something similar), so then don't need all the bootstrap stuff):
```
        return (new TwigContainer($GLOBALS['fileroot'] . "/interface/forms/soap/templates", $GLOBALS['kernel']))->getTwig()->render(
            'soap_form.twig',
            [
                "FORM_ACTION" => $GLOBALS['web_root'],
                "DONT_SAVE_LINK" => $GLOBALS['form_exit_url'],
                "CSRF_TOKEN_FORM" => CsrfUtils::collectCsrfToken(),
                "data" => $form
            ]
        );
```

then just need the following use at top of this script:
```
use OpenEMR\Common\Twig\TwigContainer;
```

That 1st parameter in TwigContainer seems a bit odd though since it seems to add the path to an array with the original path, so probably makes sense to ensure the .twig files have more specific names, such as interface/forms/soap/templates/soap_form.twig

Reply:
I agree with you but you have to remember that the C_FormSOAP.class.php file is called by the new.php file. To get away from the bootstrap file, the TwigContainer would have to injected through the new.php file. I chose not to go that route right now.